lib/pnpm-auditer.ts

```diff
@@ -27,7 +33,14 @@ async function runPnpmAudit(
 
   const arguments_ = ["audit", "--json"];
   if (registry) {
-    console.warn(yellow, "PNPM audit does not support the registry flag yet.");
+    if (pnpmAuditSupportsRegistry(pnpmVersion)) {
```

Comment:
Since we are only using the PNPM version in this branch, we can move the exec call inside the if. That way, we avoid an extra exec call unless we pass the registry flag. This is notable because the exec call takes a significant amount of time (min ~100ms).